zipkin/src/main/java/zipkin/internal/DependencyLinker.java

```diff
@@ -68,12 +68,17 @@ public DependencyLinker putTrace(Iterator<DependencyLinkSpan> spans) {
     if (logger.isLoggable(FINE)) logger.fine("traversing trace tree, breadth-first");
     for (Iterator<Node<DependencyLinkSpan>> i = tree.traverse(); i.hasNext(); ) {
       Node<DependencyLinkSpan> current = i.next();
+      DependencyLinkSpan value = current.value();
       if (logger.isLoggable(FINE)) {
-        logger.fine("processing " + current.value());
+        logger.fine("processing " + value);
+      }
+      if (value == null) {
+        logger.fine("skipping synthetic node for broken span tree");
```

Comment:
If I'm not mistaken, the only case where value is null is the case where we've added a synthetic (sentinel).
I think the code will be more clear if we show this a different way. ex. a package-private static constant `Node.MISSING_ROOT`

Then, you can compare against this reference, vs checking null.

ex. `if (current == Node.MISSING_ROOT) continue` (ps do this at the very top of this block before the processing statement)